nosetimer/plugin.py

```diff
@@ -256,7 +259,16 @@ def addFailure(self, test, err, capt=None, tb_info=None):
 
     def addSuccess(self, test, capt=None):
         """Called when a test passes."""
-        self._register_time(test, 'success')
+        time_taken = self._register_time(test, 'success')
+        if self.timer_fail is not None:
+            threshold, option = {
+                'error': (self.timer_warning, '--timer-warning'),
```

Comment:
Can we define this map only once in constructor to prevent creating it many times if timer_fail is not None and you have a lot of tests.

Reply:
@acordiner any updates here?

Reply:
Do you believe this will have a noticeable impact on performance? It is quite similar to how `_color_to_filter()` works.

The dictionary lookup could be cached in `configure()`, but it will make the code more complicated and less consistent with the way the other options are parsed. Unless you had some other approach in mind?

Reply:
Yes, I believe it will have an impact when you have a lot of tests:

```python
timer1 = timeit.Timer(
"""
t1.run()
""",
setup="""
class Test1:

    def run(self):
        return {'a': 1, 'b': 2}['a']


t1 = Test1()
"""
)
print("Test #1: %s" % timer1.timeit(10000000))


timer2 = timeit.Timer(
"""
t2.run()
""",
setup="""
class Test2:
    M = {
        'a': 1,
        'b': 2,
    }

    def run(self):
        return self.M['a']


t2 = Test2()
"""
)
print("Test #2: %s" % timer2.timeit(10000000))
```

Output:
```
Test #1: 2.29571485519
Test #2: 1.72178196907
```

It is about 25% better performance, so I think it worth to change few lines ;)

Reply:
Probably very few projects would have anything approaching 10 million unit tests. But fair enough, I can cache the dictionary to avoid the overhead of re-constructing it for every test. I am currently travelling with limited internet access, but will update the pull request as soon as I have a decent connection.

Reply:
I've made the change now. Take a look and let me know if it looks okay.